# HG changeset patch
# User Rik <rik@octave.org>
# Date 1542920554 28800
#      Thu Nov 22 13:02:34 2018 -0800
# Node ID df1cf0f7883a84aa2b75a8d53766b3b6d53c38ce
# Parent  0623e298891a39e36937a209a1d6f455291238f0
Deprecate "fontangle" property value "oblique" in version 7.

* NEWS: Announce deprecation of "oblique" value to "fontangle" property.

* graphics.in.h (text, uibuttongroup, uicontrol, uipanel, uitable): Add FIXME
comments to remove "oblique" property value in version 7.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -171,17 +171,21 @@ Summary of important user-visible change
  ** Deprecated graphics properties.
 
     The following properties or allowed corresponding values have been
     deprecated in Octave 5.0 and will be removed from Octave 7 (or
     whatever version is the second major release after 5.0):
 
       Object               | Property                | Value
       ---------------------|-------------------------|-------------------
-                           |                         |
+      text                 | fontangle               | "oblique"
+      uibuttongroup        | fontangle               | "oblique"
+      uicontrol            | fontangle               | "oblique"
+      uipanel              | fontangle               | "oblique"
+      uitable              | fontangle               | "oblique"
 
  ** The following properties or allowed corresponding values were
     deprecated in Octave 4.2 and have been removed from Octave 5.0:
 
       Object               | Property                | Value
       ---------------------|-------------------------|-------------------
       axes                 | xaxislocation           | "zero"
                            | yaxislocation           | "zero"
diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -4350,16 +4350,17 @@ public:
     // Programming note: Keep property list sorted if new ones are added.
 
     BEGIN_PROPERTIES (text)
       color_property backgroundcolor , color_property (radio_values ("{none}"), color_values (1, 1, 1))
       color_property color u , color_values (0, 0, 0)
       color_property edgecolor , color_property (radio_values ("{none}"), color_values (0, 0, 0))
       bool_property editing , "off"
       array_property extent rG , Matrix (1, 4, 0.0)
+      // FIXME: DEPRECATED: Remove "oblique" in version 7.
       radio_property fontangle u , "{normal}|italic|oblique"
       string_property fontname u , OCTAVE_DEFAULT_FONTNAME
       double_property fontsize u , 10
       radio_property fontunits SU , "inches|centimeters|normalized|{points}|pixels"
       radio_property fontweight u , "light|{normal}|demi|bold"
       radio_property horizontalalignment mu , "{left}|center|right"
       radio_property interpreter u , "{tex}|none|latex"
       radio_property linestyle , "{-}|--|:|-.|none"
@@ -5450,16 +5451,17 @@ public:
 
     BEGIN_PROPERTIES (uicontrol)
       color_property backgroundcolor , color_values (0.94, 0.94, 0.94)
       callback_property callback , Matrix ()
       array_property cdata , Matrix ()
       bool_property clipping , "on"
       radio_property enable , "{on}|inactive|off"
       array_property extent rG , Matrix (1, 4, 0.0)
+      // FIXME: DEPRECATED: Remove "oblique" in version 7.
       radio_property fontangle u , "{normal}|italic|oblique"
       string_property fontname u , OCTAVE_DEFAULT_FONTNAME
       double_property fontsize u , 10
       radio_property fontunits S , "inches|centimeters|normalized|{points}|pixels"
       radio_property fontweight u , "light|{normal}|demi|bold"
       color_property foregroundcolor , color_values (0, 0, 0)
       radio_property horizontalalignment , "left|{center}|right"
       callback_property keypressfcn , Matrix ()
@@ -5562,16 +5564,17 @@ public:
     // properties declarations.
     // Programming note: Keep property list sorted if new ones are added.
 
     BEGIN_PROPERTIES (uibuttongroup)
       color_property backgroundcolor , color_values (0.94, 0.94, 0.94)
       radio_property bordertype , "none|{etchedin}|etchedout|beveledin|beveledout|line"
       double_property borderwidth , 1
       bool_property clipping , "on"
+      // FIXME: DEPRECATED: Remove "oblique" in version 7.
       radio_property fontangle , "{normal}|italic|oblique"
       string_property fontname , OCTAVE_DEFAULT_FONTNAME
       double_property fontsize , 10
       radio_property fontunits S , "inches|centimeters|normalized|{points}|pixels"
       radio_property fontweight u , "light|{normal}|demi|bold"
       color_property foregroundcolor , color_values (0, 0, 0)
       color_property highlightcolor , color_values (1, 1, 1)
       array_property position , default_panel_position ()
@@ -5658,16 +5661,17 @@ public:
     // See the genprops.awk script for an explanation of the
     // properties declarations.
     // Programming note: Keep property list sorted if new ones are added.
 
     BEGIN_PROPERTIES (uipanel)
       color_property backgroundcolor , color_values (0.94, 0.94, 0.94)
       radio_property bordertype , "none|{etchedin}|etchedout|beveledin|beveledout|line"
       double_property borderwidth , 1
+      // FIXME: DEPRECATED: Remove "oblique" in version 7.
       radio_property fontangle , "{normal}|italic|oblique"
       string_property fontname , OCTAVE_DEFAULT_FONTNAME
       double_property fontsize , 10
       radio_property fontunits S , "inches|centimeters|normalized|{points}|pixels"
       radio_property fontweight u , "light|{normal}|demi|bold"
       color_property foregroundcolor , color_values (0, 0, 0)
       color_property highlightcolor , color_values (1, 1, 1)
       array_property position , default_panel_position ()
@@ -5755,16 +5759,17 @@ public:
       callback_property cellselectioncallback , Matrix ()
       row_vector_property columneditable , Matrix ()
       any_property columnformat S , Cell ()
       any_property columnname , "numbered"
       any_property columnwidth S , "auto"
       any_property data u , Matrix ()
       bool_property enable , "on"
       array_property extent rG , Matrix (1, 4, 0.0)
+      // FIXME: DEPRECATED: Remove "oblique" in version 7.
       radio_property fontangle u , "{normal}|italic|oblique"
       string_property fontname u , OCTAVE_DEFAULT_FONTNAME
       double_property fontsize u , 10
       radio_property fontunits S , "inches|centimeters|normalized|{points}|pixels"
       radio_property fontweight u , "light|{normal}|demi|bold"
       color_property foregroundcolor , color_values (0, 0, 0)
       callback_property keypressfcn , Matrix ()
       callback_property keyreleasefcn , Matrix ()
